File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TODOListManager.java
package Auxiliary_Modules;

public class TODOListManager {
    /* manages to do tasks.
    q1 tasks are mentioned once, and forgotten
    backup tasks are the memory of recently mentioned tasks
    * */
    private final UniqueItemSizeLimitedPriorityQueue q1 = new UniqueItemSizeLimitedPriorityQueue();
    private final UniqueItemSizeLimitedPriorityQueue backup = new UniqueItemSizeLimitedPriorityQueue();

    public TODOListManager(int todoLim) {
        q1.setLimit(todoLim);
        backup.setLimit(todoLim);
    }

    public void addTask(String e1){
        q1.add(e1);
    }
    public String getTask(){
        String temp = q1.poll();
        if(!temp.isEmpty()){backup.add(temp);}
        return temp;
    }
    public String getOldTask(){
        // task graveyard (tasks you've tackled already)
        return backup.getRNDElement();
    }
    public void clearAllTasks(){
        q1.clear();
        backup.clear();
    }
    public void clearTask(String task){
        q1.removeItem(task);
        backup.removeItem(task);
    }
    public Boolean containsTask(String task){
        return backup.contains(task);
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TODOListManager.kt
package auxiliary_modules

class TODOListManager(todoLim: Int) {
    /* manages to do tasks.
    q1 tasks are mentioned once, and forgotten
    backup tasks are the memory of recently mentioned tasks
    * */
    private val q1 = UniqueItemSizeLimitedPriorityQueue()
    private val backup = UniqueItemSizeLimitedPriorityQueue()

    init {
        q1.limit = todoLim
        backup.limit = todoLim
    }

    fun addTask(e1: String?) {
        q1.add(e1!!)
    }

    val task: String
        get() {
            val temp = q1.poll()
            if (!temp.isEmpty()) {
                backup.add(temp)
            }
            return temp
        }
    val oldTask: String
        get() =// task graveyard (tasks you've tackled already)
            backup.rNDElement

    fun clearAllTasks() {
        q1.clear()
        backup.clear()
    }

    fun clearTask(task: String) {
        q1.removeItem(task)
        backup.removeItem(task)
    }

    fun containsTask(task: String): Boolean {
        return backup.contains(task)
    }
}